runtime/lua/vim/lsp/handlers.lua

```diff
@@ -110,36 +110,45 @@ M['client/registerCapability'] = function(_, _, ctx)
 end
 
 --see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_codeAction
-M['textDocument/codeAction'] = function(_, result)
-  if result == nil or vim.tbl_isempty(result) then
+M['textDocument/codeAction'] = function(_, actions, ctx)
```

Comment:
I tried to keep the naming consistent as `result` for each handler and fully populate each handler with nil (`_`) args to make it obvious that they all have the same signature, up for discussion though.